From 6ffaf90ae97f48b7aab22aef2ee038d54bce8968 Mon Sep 17 00:00:00 2001
From: Rapaka Naveen <naveen.rapaka@intel.com>
Date: Wed, 25 Mar 2015 15:25:23 +0530
Subject: [PATCH] EM/Charger: USB OTG functionality is not working

The root cause is the PMIC Power Source Interrupt Mask Register
(MPWRSRCIRQ_REG = 0x6E0Fh)is getting masked with default value
of 0x9F because of which interrupts are masked from SOC to
PMIC for OTG detection.

Unmasked the second level Power Source Interrupt register to
trigger OTG interrupts

Change-Id: I43536eed81577295984adf98d88f8d807a2a3f6a
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1297
Signed-off-by: Rapaka Naveen <naveen.rapaka@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 11 ++++++++++-
 drivers/power/intel_pmic_ccsm.h |  3 +++
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index fd78cc7..ae52e00 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -1711,8 +1711,17 @@ static int pmic_chrgr_probe(struct platform_device *pdev)
 		}
 	}
 
-	intel_soc_pmic_writeb(chc.reg_map->pmic_mthrmirq1,
+	ret = intel_soc_pmic_writeb(chc.reg_map->pmic_mthrmirq1,
 						~MTHRMIRQ1_CCSM_MASK);
+	if (ret)
+		dev_warn(&pdev->dev, "Error writing to register: %x\n",
+				chc.reg_map->pmic_mthrmirq1);
+
+	ret = intel_soc_pmic_update(chc.reg_map->pmic_mchgrirq1,
+				MPWRSRCIRQ_CCSM_VAL, MPWRSRCIRQ_CCSM_MASK);
+	if (ret)
+		dev_warn(&pdev->dev, "Error updating register: %x\n",
+				chc.reg_map->pmic_mchgrirq1);
 
 	ret = pmic_check_initial_events();
 	if (ret)
diff --git a/drivers/power/intel_pmic_ccsm.h b/drivers/power/intel_pmic_ccsm.h
index 1ac28a4..dd0707e 100644
--- a/drivers/power/intel_pmic_ccsm.h
+++ b/drivers/power/intel_pmic_ccsm.h
@@ -240,6 +240,9 @@
 #define MTHRMIRQ1_CCSM_MASK		0x80
 #define MTHRMIRQ2_CCSM_MASK		0x3
 
+#define MPWRSRCIRQ_CCSM_MASK		0x9F
+#define MPWRSRCIRQ_CCSM_VAL		0x84
+
 /*Interrupt registers*/
 #define BATT_CHR_BATTDET_MASK	(1 << 2)
 /*Status registers*/
-- 
1.9.1

